src/lib/sidebar/scheme_data.ts

```diff
@@ -132,3 +137,22 @@ export function getUnexpectedProperties(
 
   return copy;
 }
+
+export function getFirstSchemeOrEmptyScheme(
+  schemeCollection: SchemeCollection
+): SchemeData {
+  if (schemeCollection.schemes != undefined) {
+    const schemeReferences = Object.keys(schemeCollection.schemes);
```

Comment:
Does something like `Object.keys(schemeCollection.schemes ?? {})` work, to remove the branching?